python/test/backend/compiler_inspection.py

```diff
@@ -0,0 +1,24 @@
+import os
+from triton import knobs
+
+
+def inspect_stages(self, stages, options, language, capability):
+    if os.getenv('TRITON_MOCK_INSPECT_STAGES', '0') != '0':
+        source_code = "# This is generated from Triton add_stages_inspection_hook"
+        full_name = os.path.join(knobs.cache.dump_dir, "inspect_stages.py")
+        with open(full_name, "w") as file:
+            file.write(source_code)
+    if os.getenv('TRITON_MOCK_OVERRIDE_STAGES', '0') != '0':
```

Comment:
why is this based on environment variable? Those are not great for tests since they are global

Reply:
I dropped the env vars, but I had to have a separate inspection_helper.py and inspection_helper_golden.py to do the comparison of the repro pipeline dumps.